packages/react-devtools-shared/src/backend/console.js

```diff
@@ -143,10 +154,23 @@ export function patch({
               for (const {
                 currentDispatcherRef,
                 getCurrentFiber,
+                onErrorOrWarning,
                 workTagMap,
               } of injectedRenderers.values()) {
                 const current: ?Fiber = getCurrentFiber();
                 if (current != null) {
+                  if (method === 'error' || method === 'warn') {
+                    if (typeof onErrorOrWarning === 'function') {
+                      // TODO (inline errors) The renderer is injected in two places:
+                      // 1. First by "react-devtools-shared/src/hook" which isn't stateful and doesn't supply onErrorOrWarning()
+                      // 2. Second by "react-devtools-shared/src/backend/renderer" which is and does
+                      //
+                      // We should probably move the queueing+batching mechanism from backend/renderer to this file,
+                      // so that it handles warnings logged during initial mount (potentially before step 2 above).
```

Comment:
There are two places we call the console injection, but they're used for different purposes. The one from `src/hook` is used for React Native. The one in `backend/renderer` is used for the web extension. I guess the inline package (the one also used by the test shell) is odd because it technically has both calls.

It would be nice for us to support this functionality for React Native as well.